Fail on execute AntlrKotlinTask with Gradle 8.7

I tried to run the Gradle task `AntlrKotlinTask`, but it failed.

I'm using gradle 8.7 with kotlin dsl and Java 21. I have applied plugin and add dependency. Kotlin multiplatform (JVM + JS).

Configuration cache - turn on. When cache turn off - task is always "UP-TO-DATE"



```kotlin
val generateGrammarSource by
tasks.registering(AntlrKotlinTask::class) {
//    maxHeapSize = "64m"
    arguments = arguments + listOf("-Dlanguage=Kotlin", "-visitor")
    source =
        project.objects
            .sourceDirectorySet("antlr", "antlr")
            .srcDir("src/commonAntlr/antlr")
            .apply { include("*.g4") }

    val pkgName = "com.strumenta.antlrkotlin.parsers.generated"
    packageName = pkgName

    // outputDirectory is required, put it into the build directory
    // if you do not want to add the generated sources to version control
    outputDirectory = File("build/generated-src/commonAntlr/kotlin")
}
```


Gradle Error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache state could not be cached: field `logger` of `org.gradle.internal.logging.source.JavaUtilLoggingSystem` bean found in field `loggingSystem` of `org.gradle.internal.logging.services.DefaultLoggingManager$StartableLoggingSystem` bean found in field `javaUtilLoggingSystem` of `org.gradle.internal.logging.services.DefaultLoggingManager` bean found in field `loggingManager` of `org.gradle.process.internal.worker.DefaultWorkerProcessFactory` bean found in field `workerProcessBuilderFactory` of task `:common:expression:generateGrammarSource` of type `com.strumenta.antlrkotlin.gradle.AntlrKotlinTask`: error writing value of type 'java.util.logging.LogManager$RootLogger'
> Unable to make field final java.util.logging.LogManager java.util.logging.LogManager$RootLogger.this$0 accessible: module java.logging does not "opens java.util.logging" to unnamed module @67c339bb
```

I've tried to open all necessary java modules.

```  
--add-opens=java.logging/java.util.logging=ALL-UNNAMED --add-opens java.locks/java.util.concurrent.locks=ALL-UNNAMED --add-opens java.base/java.util.concurrent.locks=ALL-UNNAMED --add-opens java.base/java.lang.ref=ALL-UNNAMED --add-opens java.base/java.text=ALL-UNNAMED --add-opens java.base/sun.util.calendar=ALL-UNNAMED --add-opens java.base/java.util.concurrent=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED
```

There is another error: 
```
Could not load the value of field `m` of `java.util.Collections$SynchronizedMap` bean found in field `listeners` of `java.util.logging.LogManager` bean found in field `this$0` of `java.util.logging.LogManager$RootLogger` bean found in field `logger` of `org.gradle.internal.logging.source.JavaUtilLoggingSystem` bean found in field `loggingSystem` of `org.gradle.internal.logging.services.DefaultLoggingManager$StartableLoggingSystem` bean found in field `javaUtilLoggingSystem` of `org.gradle.internal.logging.services.DefaultLoggingManager` bean found in field `loggingManager` of `org.gradle.process.internal.worker.DefaultWorkerProcessFactory` bean found in field `workerProcessBuilderFactory` of task `:common:expression:generateGrammarSource` of type `com.strumenta.antlrkotlin.gradle.AntlrKotlinTask`.
```

Gradle: configuration cache report in attachment (Github not allow to attach html, so please, change file type to html to view report). If you need any details, please let me know.

[configuration-cache-report.log](https://github.com/Strumenta/antlr-kotlin/files/14873826/configuration-cache-report.log)

--- a/antlr-kotlin-gradle-plugin/src/main/kotlin/com/strumenta/antlrkotlin/gradle/AntlrKotlinTask.kt
+++ b/antlr-kotlin-gradle-plugin/src/main/kotlin/com/strumenta/antlrkotlin/gradle/AntlrKotlinTask.kt
@@ -23,11 +23,7 @@ import javax.inject.Inject
  */
 @NonNullApi
 @CacheableTask
-public abstract class AntlrKotlinTask @Inject constructor(
-  private val deleter: Deleter,
-  private val projectLayout: ProjectLayout,
-  private val workerProcessBuilderFactory: WorkerProcessFactory,
-) : SourceTask() {
+public abstract class AntlrKotlinTask : SourceTask() {
   private var sourceSetDirectories: FileCollection? = null
 
   /**
@@ -99,6 +95,19 @@ public abstract class AntlrKotlinTask @Inject constructor(
   @get:PathSensitive(PathSensitivity.RELATIVE)
   @get:InputFiles
   protected val stableSources: FileCollection = project.files({ this.source })
+
+  // Note(Edoardo): we had to move injected properties from
+  //  constructor injection to method injection to correctly
+  //  support Gradle's configuration cache
+
+  @get:Inject
+  protected abstract val deleter: Deleter
+
+  @get:Inject
+  protected abstract val projectLayout: ProjectLayout
+
+  @get:Inject
+  protected abstract val workerProcessBuilderFactory: WorkerProcessFactory
 
   /**
    * Generate the parsers.

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -1,13 +1,13 @@
 [versions]
-kotlin = "1.9.22"
-kotlin-wrappers = "1.0.0-pre.683"
+kotlin = "1.9.23"
+kotlin-wrappers = "1.0.0-pre.729"
 kotlinx-io = "0.3.1"
 kotlinx-benchmark = "0.4.10"
-kotlinx-resources = "0.4.0"
+kotlinx-resources = "0.4.1"
 antlr4 = "4.13.1"
-dokka = "1.9.10"
+dokka = "1.9.20"
 researchgate-release = "3.0.2"
-maven-publish = "0.27.0"
+maven-publish = "0.28.0"
 gradle-plugin-publish = "1.2.1"
 
 [libraries]

